### PR TITLE
Wrap PyTorch multinomial distribution for use in Pyro

### PR DESCRIPTION
**Pull request by fritzo**

### CODE REVIEW

File: pyro/distributions/torch/multinomial.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+@copy_docs_from(_Multinomial)
+class Multinomial(TorchDistribution):
+    enumerable = True
+
+    def __init__(self, ps, n, *args, **kwargs):
+        if isinstance(n, Variable):
+            n = n.data
+        if not isinstance(n, numbers.Number):
+            if n.max() != n.min():
```
Comment (neerajprad) [inline pyro/distributions/torch/multinomial.py]: Do you think we should change our API to have `n` be `int` as well?

Comment (fritzo) [reply pyro/distributions/torch/multinomial.py]: That seems reasonable. I'd like to switch all our distributions to look more like PyTorch distributions. I'd also like to eliminate `.log_pdf()` and rename `.batch_log_pdf()` to `.log_prob()`.

Comment (neerajprad) [reply pyro/distributions/torch/multinomial.py]: I want to start making these changes on the Pyro side too. Removing all Pyro distributions (that do not have a corresponding PyTorch implementation) will probably help make this faster. What do you think?

Comment (fritzo) [reply pyro/distributions/torch/multinomial.py]: I'm a bit wary of removing all Pyro distributions. I think we may want to keep Pyro versions of `Delta`, `Empirical`, and `MultivariateNormal` long-term. The other Pyro-only distributions are currently being worked on by various PyTorch folks, so I think it's safe to remove those.

**Pull request merged**